Source/WebCore/platform/graphics/egl/GLContextEGL.cpp

```diff
@@ -161,9 +178,43 @@ bool GLContextEGL::getEGLConfig(EGLDisplay display, EGLConfig* config, EGLSurfac
 
 std::unique_ptr<GLContextEGL> GLContextEGL::createWindowContext(GLNativeWindowType window, PlatformDisplay& platformDisplay, EGLContext sharingContext)
 {
+    Function<bool(int)> checkCompatibleVisuals = nullptr;
+#if PLATFORM(X11)
+    XUniquePtr<XVisualInfo> visualInfo;
+    if (platformDisplay.type() == PlatformDisplay::Type::X11) {
+        Display* x11Display = downcast<PlatformDisplayX11>(platformDisplay).native();
+        XWindowAttributes attributes;
+        if (!XGetWindowAttributes(x11Display, static_cast<Window>(window), &attributes))
+            return nullptr;
+
+        auto visualInfoForID = [x11Display](VisualID visualID) -> XUniquePtr<XVisualInfo> {
+            XVisualInfo templateVisualInfo;
+            templateVisualInfo.visualid = visualID;
+            int visualInfoCount;
+            return XUniquePtr<XVisualInfo>(XGetVisualInfo(x11Display, VisualIDMask, &templateVisualInfo, &visualInfoCount));
+        };
+
+        visualInfo = visualInfoForID(XVisualIDFromVisual(attributes.visual));
+        if (!visualInfo)
+            return nullptr;
+
+        checkCompatibleVisuals = [&visualInfo, &visualInfoForID](int configVisualID) {
```

Comment:
And visualInfo = WTFMove(visualInfo) here. And we should do the same for the visualInfoFoID lambda, I guess. Because it's declared inside the if and passed to the Function as reference, but released when getEGLConfig is called.